lib/modelManagers/reportModel.js

```diff
@@ -0,0 +1,140 @@
+const {
+  Reports,
+  User,
+  Articles,
+  ReportTag
+} = require('../../database/models');
+
+/**
+ * @description handle reporting articles
+ *
+ * @class ReportModelManager
+ */
+class ReportModelManager {
+/**
+ * @description create a new report
+ *
+ * @static
+ * @param {any} reportData Report data to be created
+ * @returns {null} no return
+ *
+ * @memberOf ReportModelManager
+ */
+  static async createReport(reportData) {
+    const createdReport = await Reports.create({
+      userId: reportData.userId,
+      articleId: reportData.articleId,
+      reportTagId: reportData.reportTagId,
+      message: reportData.message
+    });
+    return createdReport;
+  }
+
+  /**
+ * @description get all reports
+ *
+ * @static
+ * @param {any} limit pagination limit
+ * @param {any} offset pagination offset
+ * @returns {Object} return all reports
+ *
+ * @memberOf ReportModelManager
+ */
+  static async getAllReports(limit, offset) {
+    const allReports = await Reports.findAll({
+      limit,
+      offset,
+      include: [
+        {
+          model: User,
+          as: 'reporter',
+          attributes: ['firstname', 'lastname']
+        },
+        {
+          model: ReportTag,
+          as: 'tag',
+          attributes: ['name']
+        },
+
+      ]
+    });
+    return allReports;
+  }
+
+  /**
+   * @description get a single report
+   *
+   * @static
+   * @param {any} id article id
+   * @returns {Object} return a single report
+   *
+   * @memberOf ReportModelManager
+   */
+  static async getReportById(id) {
+    const getAReport = await Reports.findOne({
+      where: {
+        id
+      },
+      include: [
+        {
+          model: User,
+          as: 'reporter',
+          attributes: ['firstname', 'lastname']
+        },
+        {
+          model: Articles,
+          as: 'userArticle',
+          attributes: ['title', 'description', 'content']
+        },
+        {
+          model: ReportTag,
+          as: 'tag',
+          attributes: ['name']
+        }
+      ]
+    });
+    return getAReport;
+  }
+
+  /**
+ * @description - handles report update
+ *
+ * @static
+ * @param {any} id Report id to be updated
+ * @returns {object} Return updated report
+ *
+ * @memberOf ReportTagModelManager
+ */
+  static async updateReport(id) {
+    const updated = await Reports.update(
+      {
+        resolved: true
```

Comment:
Can we not hardcode `resolved: true` here. Doing it this way will make it impossible to change `resolved` to false.

Reply:
Fixed